lib/win32/certstore/certificate/cert_base.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Author:: Piyush Awasthi (<piyush.awasthi@msystechnologies.com>)
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'store/crypto'
+require 'store/assertions'
+
+module Win32
+  class Certstore
+    class Certificate
+      module CertBase
+        include Win32::Store::Assertions
+        include Win32::Store::Crypto
+        include Chef::Mixin::ShellOut
+        include Chef::Mixin::WideString
+
+        def list_cert(store_handle)
```

Comment:
I don't think `list_cert` belongs here. Since this is the `Certificate` class I would expect this to represent a single certificate instance. I'd expect to list certs using a `CertStore`